src/sinks/gcp/pubsub.rs

```diff
@@ -306,6 +306,34 @@ mod integration_tests {
         config(topic).build(cx).await.expect("Building sink failed")
     }
 
+    #[tokio::test]
+    async fn publish_metrics() {
+        trace_init();
+
+        let (topic, subscription) = create_topic_subscription().await;
+        let (sink, healthcheck) = config_build(&topic).await;
+
+        healthcheck.await.expect("Health check failed");
+
+        let (batch, mut receiver) = BatchNotifier::new_with_receiver();
+        let (input, events) = metrics_with_stream(100, Some(batch));
+        run_and_assert_sink_compliance(sink, events, &HTTP_SINK_TAGS).await;
+        assert_eq!(receiver.try_recv(), Ok(BatchStatus::Delivered));
+
+        let response = pull_messages(&subscription, 1000).await;
+        let messages = response
+            .receivedMessages
+            .as_ref()
+            .expect("Response is missing messages");
+        assert_eq!(input.len(), messages.len());
+        for i in 0..input.len() {
+            let data = messages[i].message.decode_data();
```

Comment:
I missed here that `decode_data` expects the message to have be a  specific structure with a `timestamp` and `message` field. The metrics received will not have this structure. Instead I would suggest adding another function to `PullMessage` that just returns a `serde_json::Value`:

```
        fn decode_data_as_value(&self) -> Value {
            let data = BASE64_STANDARD
                .decode(&self.data)
                .expect("Invalid base64 data");
            let data = String::from_utf8_lossy(&data);
            serde_json::from_str(&data).expect("Invalid json")
        }
```

```suggestion
            let data = messages[i].message.decode_data_as_value();
```